storage/replica_command.go

```diff
@@ -1477,29 +1480,38 @@ func (r *Replica) sha512(desc roachpb.RangeDescriptor, snap engine.Engine) ([]by
 	for ; iter.Valid(); iter.Next() {
 		key := iter.Key()
 		value := iter.Value()
+
+		if log.V(1) {
```

Comment:
Rather than test `log.V(1)` inside the loop, I think it would be better to initialize `debug` based on `log.V(1)` outside the loop and change the above test to `if debug != nil {`. The `debug` initialization would look like:

```
  var debug *roachpb.VerifyChecksumRequest_Debug
  if log.V(1) {
    debug = &roachpb.VerifyChecksumRequest_Debug{}
  }
```

Then at the end of this method you can just return `debug` without having to zero it if `!log.V(1)`.